OSDOCS-3146: Note about AWS us-east-1 region.

EricPonvelle opened the pull request:
This PR adds a note to the ROSA documentation, alerting users that AWS `us-east-1` is required for OIDC providers.

JIRA: https://issues.redhat.com/browse/OSDOCS-3146

PREVIEW: 

* **STS** - https://deploy-preview-40537--osdocs.netlify.app/openshift-rosa/latest/rosa_getting_started_sts/rosa-sts-aws-prereqs.html#rosa-aws-prereqs_rosa-sts-aws-prerequisites

arendej reviewed (approved):
looks good!

arendej commented:
> @arendej OIDC providers should only apply to STS clusters, right? Standard IAM clusters should not require OIDC providers. In addition, shouldn't we clarify what kind of OIDC providers we're talking about (specifically that a customer's OIDC identity provider doesn't require us-east-1, but the STS OIDC provider does).


Yes, accurate. Thank you @michaelryanmcneill!
- ROSA's AWS OIDC provider only applies to ROSA STS clusters.
- ROSA's AWS OIDC provider currently still relies on us-east-1 (RFE in the works about this) and should be called out.

michaelryanmcneill commented:
@EricPonvelle we should reopen this to fix the wording because this can be confusing to customers.

EricPonvelle commented:
@michaelryanmcneill and @arendej I updated the text on the STS side. I also removed the note from the non-STS install path. Please take a look at your earliest convenience.

michaelryanmcneill commented:
lgtm

Thanks @EricPonvelle!

mjpytlak commented on rosa_getting_started_sts/rosa-sts-aws-prereqs.adoc:
```diff
@@ -11,6 +11,12 @@ toc::[]
+
+[IMPORTANT]
+====
+When you create a ROSA cluster using AWS STS, an associated AWS OpenID Connect (OIDC) identity provider is created as well. This OIDC provider configuration relies on a public key that is located in the `us-east-1` AWS region. Customers with AWS SCPs must allow the use of the `us-east-1` AWS region even if these clusters are deployed in a different region.
```
```suggestion
When you create a ROSA cluster using AWS STS, an associated AWS OpenID Connect (OIDC) identity provider is created as well. This OIDC provider configuration relies on a public key that is located in the `us-east-1` AWS region. Customers with AWS SCPs must allow the use of the `us-east-1` AWS region, even if these clusters are deployed in a different region.
```

EricPonvelle commented:
/cherry-pick enterprise-4.9

EricPonvelle commented:
/cherry-pick enterprise-4.10
